be/CMakeLists.txt

```diff
@@ -107,6 +107,20 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         set(CXX_GCC_FLAGS "${CXX_GCC_FLAGS} -Werror")
     endif()
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    # When LLVM is used, should give GCC_HOME to get c++11 header to use new string and list
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "16.0.0")
+        if (DEFINED ENV{STARROCKS_GCC_INSTALL_DIR})
+            set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} --gcc-install-dir=$ENV{STARROCKS_GCC_INSTALL_DIR}")
```

Comment:
have tried, it still works the same way as gcc-toolchain in clang16. this gcc-install-dir looks like unnecessary.

Reply:
gcc-toolchain also use gcc executes, such as ld, but gcc-install-dir just use header files and libraries.

Reply:
and also they have differenct argument.

```
[root@7c13b927fad8 be]# clang -v --gcc-toolchain='/opt/rh/gcc-toolset-10/root/usr'
Ubuntu clang version 16.0.6 (++20230610113324+7cbf1a259152-1~exp1~20230610233415.100)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /opt/ldb/bin
Found candidate GCC installation: /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0
Selected GCC installation: /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0
Candidate multilib: .;@m64
Selected multilib: .;@m64


[root@7c13b927fad8 be]# clang -v --gcc-install-dir='/opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0'
Ubuntu clang version 16.0.6 (++20230610113324+7cbf1a259152-1~exp1~20230610233415.100)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /opt/ldb/bin
Selected GCC installation: /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

Reply:
didn't see the benefit. With the correct gcc toolchain provided, I can build the project with clang16 with no error (except the additional Werror change).

Reply:
this depends on your environment, i just install ldb based on centos develop docker and got errors as described above.

and copied as following.

```
/opt/ldb/bin/x86_64-pc-linux-gnu-ld: cannot find -lstdc++
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

and also maybe in your environment the linker is still ld instead of ldb which is faster than ld.

Reply:
no. I replaced ld by lld. w.r.t to gcc-install-dir v.s. gcc-toolchain, the former one is good to force selecting a gcc installation no matter its version is the largest or not, while the gcc-toolchain can auto detect multiple gcc installation and automatically choose the largest version.

Given the single installation of gcc, or the intended gcc to be used is the largest one. There is no significant difference.

Maybe I am reading wrong from llvm doc about these two options.